Scrape-Subreddit-Users-master/GetSubredditSubscribersComments.py

```diff
@@ -35,16 +36,15 @@
             pass
         else:
             session.add(User(username))
+            session.commit()
     except Exception as err:
         # TODO: determine exact SQLAlchemy error thrown here
         print("could not save user %s: %s" % (username, err))
 
-session.commit()
 session = Session()
 
-from sqlalchemy import distinct
-allusers = session.query(User.username)
-scraped = session.query(distinct(Comment.author))
+allusers = session.query(User.username).all()
+scraped = session.query(distinct(Comment.author)).all()
```

Comment:
In sqlalchemy, after you query, you have to retrieve the result out of the query object.